integration_tests/regression/test/issue_2142_test.dart

```diff
@@ -0,0 +1,15 @@
+@Skip('https://github.com/dart-lang/test/issues/2142')
+library;
+
+import 'package:test/test.dart';
+
+// ignore: avoid_relative_lib_imports
+import '../lib/issue_2142/import.dart';
```

Comment:
Note that this one probably would be broken either way?

Reply:
Yes, I think we should not attempt to support `../lib`. I'd be happy to check for this and block it if it's easy.